G1ANT.Addon.Net/Commands/ImapGetEmailsCommand.cs

```diff
@@ -0,0 +1,166 @@
+﻿/**
+*    Copyright(C) G1ANT Ltd, All rights reserved
+*    Solution G1ANT.Addon, Project G1ANT.Addon.Net
+*    www.g1ant.com
+*
+*    Licensed under the G1ANT license.
+*    See License.txt file in the project root for full license information.
+*
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using G1ANT.Language;
+using G1ANT.Language.Models;
+using MailKit;
+using MailKit.Search;
+
+namespace G1ANT.Addon.Net
+{
+    [Command(Name = "imap.getmails", Tooltip = "This command uses the IMAP protocol to check an email inbox and allows the user to analyze their messages received within a specified time span, with the option to consider only unread messages and/or mark all of the checked ones as read")]
+    public class ImapGetMailsCommand : Command
+    {
+        public class Arguments : CommandArguments
+        {
+            [Argument(Required = true, Tooltip = "Folder to fetch emails from")]
+            public TextStructure Folder { get; set; } = new TextStructure("INBOX");
+
+            [Argument(Required = true, Tooltip = "How many emails to download")]
+            public IntegerStructure Count { get; set; } = new IntegerStructure(50);
+
+            [Argument(Required = false, Tooltip = "Start downloading emails newer than a specified one")]
+            public SimplifiedMessageSummary FromEmail { get; set; }
+
+            [Argument(Required = false, Tooltip = "Starting date for messages to be checked")]
+            public DateStructure SinceDate { get; set; }
+
+            [Argument(Required = false, Tooltip = "Ending date for messages to be checked")]
+            public DateStructure ToDate { get; set; } = new DateStructure(DateTime.Now);
+
+            [Argument(Required = false, Tooltip = "If set to `true`, only unread messages will be checked")]
+            public BooleanStructure OnlyUnreadMessages { get; set; } = new BooleanStructure(false);
+
+            [Argument(Required = false, Tooltip = "Mark analyzed messages as read")]
+            public BooleanStructure MarkAsRead { get; set; } = new BooleanStructure(true);
+
+            [Argument(Required = false, Tooltip = "Name of a list variable where the returned mail variables will be stored")]
```

Comment:
I believe all required tags are redundant, especially when you provide command with default value of argument